source/kernel/src/registry/mod.rs

```diff
@@ -455,7 +466,29 @@ impl Registry {
 
 // Envelope
 
+impl<P> OpenEnvelope<P> {
+    pub fn fill(self, contents: P) -> Envelope<P> {
```

Comment:
nit, take it or leave it: is there a reason we can't get rid of the `P` generic on `OpenEnvelope` and just have 
```suggestion
impl OpenEnvelope {
    pub fn fill<P>(self, contents: P) -> Envelope<P> {
```
this might reduce some unneeded monomorphization possibly?

Reply:
Gunna defer on this because I want to revisit this and move the whole API over to this to get rid of the old bad one.